sequitur.go

```diff
@@ -323,21 +337,60 @@ func (g *Grammar) PrettyPrint(w io.Writer) error {
 // ErrAlreadyParsed is returned if the grammar instance has already parsed a grammar
 var ErrAlreadyParsed = errors.New("sequitor: grammar already parsed")
 
-// Parse parses a byte string
-func (g *Grammar) Parse(str []byte) error {
+// ErrEmptyInput is returned if the input string is empty
+var ErrEmptyInput = errors.New("sequitor: empty input")
+
+// ErrMalformedUTF8 is returned if the input string contains malformed UTF-8
+var ErrMalformedUTF8 = errors.New("sequitor: malformed utf-8 input")
+
+// ParseUTF8 parses a byte string using utf-8 encoding
+func ParseUTF8(str []byte) (*Grammar, error) {
+	g := new(Grammar)
+	return g, g.parse(str, true)
+}
+
+// ParseBinary parses a binary byte string
+func ParseBinary(str []byte) (*Grammar, error) {
+	g := new(Grammar)
+	return g, g.parse(str, false)
+}
+
+func (g *Grammar) parse(str []byte, useUTF8 bool) error {
 	if g.base != nil {
 		return ErrAlreadyParsed
 	}
+	if len(str) == 0 {
```

Comment:
Is this actually needed? ISTM that an empty input probably should be valid, and having no rules in the grammar should be OK too (or just one rule with an empty production).

Reply:
I agree, but will leave this to @dgryski now that he has merged this PR.